frontend/pages/admin/campaigns/create.tsx

```diff
@@ -0,0 +1,182 @@
+      .integer('Coupon denomination must be an integer.')
+      .positive('Coupon denomination must be a positive.')
+      .min(10, 'Coupon denomination must be at least $10')
+      .when('promisedAmount', (promisedAmount, schema) => {
+        return schema.test({
+          test: (couponDenomination: number) =>
+            canBecomeInteger(couponDenomination) && canBecomeInteger(promisedAmount)
+              ? promisedAmount % couponDenomination === 0
+              : true,
+          message: `Coupon denomination must be a factor of $${promisedAmount}.`,
+        });
+      }),
+    start: Yup.date()
+      .required('Start date is required.')
+      .typeError('Start date must be a date.')
+      .when('end', (endDate, schema) =>
+        isValidDate(endDate) ? schema.max(endDate, 'Start date cannot be after End date') : schema,
+      )
+      .min(moment().endOf('day'), 'Start date cannot be today or in the past.'),
+    end: Yup.date()
+      .required('End date is required.')
+      .typeError('End date must be a date.')
+      .when('startDate', (startDate, schema) =>
+        isValidDate(startDate) ? schema.min(startDate, 'End date cannot be before Start date') : schema,
+      )
+      .min(moment().endOf('day'), 'End date cannot be today or in the past.'),
+    imageBase64: Yup.string().required('Campaign image is required.'),
+    interestId: Yup.number().nullable().required(),
+    charities: Yup.array()
+      .of(
+        Yup.object().shape({
+          charity: Yup.object()
+            .shape({
+              id: Yup.number().required('Charity is required'),
+            })
+            .required('Charity is required'),
+          givingSgUrl: Yup.string()
+            .required('Giving.sg campaign url is required')
+            .url('A valid url including http:// or https:// is required'),
+        }),
+      )
+      .required('At least one charity is required.')
+      .min(1, 'At least one charity is required.')
+      .max(MAX_NUM_OF_CAMPAIGN_CHARITIES, `There can only be a maximum of ${MAX_NUM_OF_CAMPAIGN_CHARITIES} charities.`),
+    primaryDonor: Yup.object({
+      name: Yup.string().required('Primary donor name is required.'),
+      email: Yup.string().required('Primary donor email is required.').email('Donor email is not in the correct form.'),
+    }).required('Primary donor is required'),
+  },
+  [
+    ['start', 'end'],
+    ['promisedAmount', 'couponDenomination'],
+  ],
+);
+
+const CampaignCreate = () => {
+  const router = useRouter();
+  const { interestId } = router.query;
+  const { data: interest } = useSWR<Nullable<Interest>>(`${InterestsAPI.INTERESTS_URL}/getInterest`, () =>
+    isInteger(interestId) ? api.interests.getInterest(Number(interestId)).then((r) => r.payload) : null,
+  );
+  const [initialValues, setInitialValues] = useState<CampaignFormData>({
+    start: null,
+    end: null,
+    interestId: null,
+    charities: [{}],
+  });
+
+  useEffect(() => {
+    if (interest) {
```

Comment:
Does this ensure that the useSWR api request finishes before this?

Reply:
What's the problem if it doesn't?

Reply:
Shouldn't the page show loading if there is a interest api request being made?

Reply:
It can just show the empty form? It's a bit weird to show loading for a form. But I can add it if you think it's better.

Reply:
> bit weird to show loading for a form

Okay I agree. Can leave it at it is